pkg/sdk/users_integration_test.go

```diff
@@ -0,0 +1,240 @@
+			With: Bool(true),
+			Tags: tags,
+		}
+		err := client.Users.Create(ctx, id, opts)
+		require.NoError(t, err)
+		userDetails, err := client.Users.Describe(ctx, id)
+		require.NoError(t, err)
+		assert.Equal(t, id.name, userDetails.Name.Value)
+		assert.Equal(t, strings.ToUpper(loginName), userDetails.LoginName.Value)
+
+		user, err := client.Users.Show(ctx, &ShowUserOptions{
+			Like: &Like{
+				Pattern: &id.name,
+			},
+		})
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(user))
+		assert.Equal(t, id.name, user[0].Name)
+	})
+
+	t.Run("test if not exists", func(t *testing.T) {
+		id := randomAccountObjectIdentifier(t)
+		tagValue := randomString(t)
+		tags := []TagAssociation{
+			{
+				Name:  NewAccountObjectIdentifier(tag.Name),
+				Value: tagValue,
+			},
+		}
+		password := randomString(t)
+		loginName := randomString(t)
+
+		opts := &CreateUserOptions{
+			IfNotExists: Bool(true),
+			name:        id,
+			ObjectProperties: &UserObjectProperties{
+				Password:  &password,
+				LoginName: &loginName,
+			},
+			ObjectParameters: &UserObjectParameters{
+				EnableUnredactedQuerySyntaxError: Bool(true),
+			},
+			SessionParameters: &SessionParameters{
+				Autocommit: Bool(true),
+			},
+			With: Bool(true),
+			Tags: tags,
+		}
+		err := client.Users.Create(ctx, id, opts)
+		require.NoError(t, err)
+		userDetails, err := client.Users.Describe(ctx, id)
+		require.NoError(t, err)
+		assert.Equal(t, id.name, userDetails.Name.Value)
+		assert.Equal(t, strings.ToUpper(loginName), userDetails.LoginName.Value)
+
+		user, err := client.Users.Show(ctx, &ShowUserOptions{
+			Like: &Like{
+				Pattern: &id.name,
+			},
+		})
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(user))
+		assert.Equal(t, id.name, user[0].Name)
+	})
+
+	t.Run("test no options", func(t *testing.T) {
+		id := randomAccountObjectIdentifier(t)
+
+		err := client.Users.Create(ctx, id, nil)
+		require.NoError(t, err)
+		userDetails, err := client.Users.Describe(ctx, id)
+		require.NoError(t, err)
+		assert.Equal(t, id.name, userDetails.Name.Value)
+		assert.Equal(t, strings.ToUpper(id.name), userDetails.LoginName.Value)
+		assert.Empty(t, userDetails.Password.Value)
+
+		user, err := client.Users.Show(ctx, &ShowUserOptions{
+			Like: &Like{
+				Pattern: &id.name,
+			},
+		})
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(user))
+		assert.Equal(t, id.name, user[0].Name)
+	})
+}
+
+func TestInt_UserDescribe(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	user, userCleanup := createUser(t, client)
+	t.Cleanup(userCleanup)
+
+	t.Run("when user exists", func(t *testing.T) {
+		userDetails, err := client.Users.Describe(ctx, user.ID())
+		require.NoError(t, err)
+		assert.Equal(t, user.Name, userDetails.Name.Value)
+	})
+
+	t.Run("when user does not exist", func(t *testing.T) {
+		id := NewAccountObjectIdentifier("does_not_exist")
+		_, err := client.Users.Describe(ctx, id)
+		assert.ErrorIs(t, err, ErrObjectNotExistOrAuthorized)
```

Comment:
One thing I find quite annoying is when we raise an error like this, we get no information about where the error was initially raised. Since errors are regular values in Go, they don't carry any stack trace with them.
I think we might consider 
1) using a third-party library(https://github.com/go-errors/errors)
2) converting those errors into a thunk function type, like this:
```
	ErrObjectNotExistOrAuthorized = func() error {
		return errors.New("object does not exist or not authorized")
	}
```
that way we can add a breakpoint inside it and see the stack trace with a debugger
3) add some basic info to the error, like what kind of object does not exist etc.

Reply:
I like 1) and 2).

Reply:
perhaps wrapping the errors at each place where they are thrown. that way you can see the stacktrace as part of the error message as a sort of linked list
`error: <err msg> occurred in <func> -> <struct> -> <other func> -> <other struct>`

Reply:
Right, @sfc-gh-jcieslak and I were discussing an option like this, but it requires quite a lot of effort when rethrowing errors everywhere. We need to also agree what is it that we want to achieve here.
My main pain point was not knowing when the error initially occurred during debugging a failing test.
I'm not sure what functionality can an external library provide, but we can either:
a) use the suggestion from point 2)
b) add a trace using runtime package when the error initially enters our application(i.e when we create one ourselves or get one from an external library)
I find such information more useful in this case because I can just go directly to the line when the error was raised.
Having a wrapper for errors will still be useful, but it:
a) needs additional effort on every rethrow
b) still requires a bit of decoding if we want to go to a specific line when the error happened.
Maybe such an error would be more helpful for the customer? I don't know, but I think those 2 approaches are not mutually exclusive, but whether we need the latter if we have the former is something is something I'm 100% sure about.